src/Appwrite/Platform/Workers/Webhooks.php

```diff
@@ -117,10 +133,72 @@ private function execute(array $events, string $payload, Document $webhook, Docu
-            $this->errors[] = \curl_error($ch) . ' in events ' . implode(', ', $events) . ' for webhook ' . $webhook->getAttribute('name');
-        }
-
+        $responseBody = \curl_exec($ch);
+        $curlError = \curl_error($ch);
+        $statusCode = curl_getinfo($ch, CURLINFO_RESPONSE_CODE);
         \curl_close($ch);
+
+        if (!empty($curlError) || $statusCode >= 400) {
+            $dbForConsole->increaseDocumentAttribute('webhooks', $webhook->getId(), 'attempts', 1);
+            $webhook = $dbForConsole->getDocument('webhooks', $webhook->getId());
+            $attempts = $webhook->getAttribute('attempts');
+
+            $logs = '';
+            $logs .= 'URL: ' . $webhook->getAttribute('url') . "\n";
+            $logs .= 'Method: ' . 'POST' . "\n";
+
+            if (!empty($curlError)) {
+                $logs .= 'CURL Error: ' . $curlError . "\n";
+                $logs .= 'Events: ' . implode(', ', $events) . "\n";
+            } else {
+                $logs .= 'Status code: ' . $statusCode . "\n";
+                $logs .= 'Body: ' . "\n" . \mb_strcut($responseBody, 0, 10000) . "\n"; // Limit to 10kb
+            }
+
+            $webhook->setAttribute('logs', $logs);
+
+            if ($attempts >= self::MAX_FAILED_ATTEMPTS) {
+                $webhook->setAttribute('enabled', false);
+
+                $protocol = App::getEnv('_APP_OPTIONS_FORCE_HTTPS') == 'disabled' ? 'http' : 'https';
+                $hostname = App::getEnv('_APP_DOMAIN');
+                $projectId = $project->getId();
+                $webhookId = $webhook->getId();
+
+                $template = __DIR__ . '/../../../../app/config/locale/templates/email-webhook.phtml';
+                $template = new View($template);
+
+                $template->setParam('user', $user->getAttribute('name'));
+                $template->setParam('webhook', $webhook->getAttribute('name'));
+                $template->setParam('project', $project->getAttribute('name'));
+                $template->setParam('url', $webhook->getAttribute('url'));
+                $template->setParam('error', $curlError ??  'The server returned ' . $statusCode . ' status code');
+                $template->setParam('redirect', $protocol . '://' . $hostname . "/console/project-$projectId/settings/webhooks/$webhookId");
+                $template->setParam('attempts', $attempts);
+
+                $subject = 'Webhook deliveries have been paused';
+                $body = Template::fromFile(__DIR__ . '/../../../../app/config/locale/templates/email-base-branded.tpl');
+
+                $body
+                    ->setParam('{{subject}}', $subject)
+                    ->setParam('{{message}}', $template->render())
+                    ->setParam('{{year}}', date("Y"));
+
+                $queueForMails
+                    ->setSubject($subject)
+                    ->setBody($body->render())
+                    ->setRecipient($user->getAttribute('email'))
```

Comment:
User comes from `$user = new Document($payload['user'] ?? []);`.
Who is user in payload of webhook? 🤔 Most logical approach would be to send email to all members of organization, right?